tests/lib/ifaces/linux_bridge_iface_test.py

```diff
@@ -0,0 +1,420 @@
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+
+from copy import deepcopy
+
+import pytest
+
+from libnmstate.error import NmstateValueError
+from libnmstate.schema import Interface
+from libnmstate.schema import InterfaceState
+from libnmstate.schema import LinuxBridge as LB
+
+from libnmstate.ifaces.linux_bridge import LinuxBridgeIface
+from libnmstate.ifaces.ifaces import Ifaces
+
+from ..testlib.bridgelib import LINUX_BRIDGE_IFACE_NAME
+from ..testlib.bridgelib import TEST_SLAVE_NAMES
+from ..testlib.bridgelib import SLAVE1_IFACE_NAME
+from ..testlib.bridgelib import SLAVE1_PORT_CONFIG
+from ..testlib.bridgelib import SLAVE2_IFACE_NAME
+from ..testlib.bridgelib import SLAVE2_PORT_CONFIG
+from ..testlib.bridgelib import SLAVE2_VLAN_CONFIG_TRUNK
+from ..testlib.bridgelib import TRUNK_TAGS_ID_RANGES
+from ..testlib.bridgelib import gen_bridge_iface_info
+from ..testlib.bridgelib import gen_bridge_iface_info_with_vlan_filter
+from ..testlib.ifacelib import gen_foo_iface_info
+
+Port = LB.Port
+Vlan = LB.Port.Vlan
+
+
+class TestLinuxBridgeIface:
+    def test_linux_bridge_sort_slaves(self):
+        iface_info1 = gen_bridge_iface_info()
+        iface_info2 = gen_bridge_iface_info()
+        iface_info2[LB.CONFIG_SUBTREE][LB.PORT_SUBTREE].reverse()
+
+        assert (
+            LinuxBridgeIface(iface_info1).to_dict()
+            == LinuxBridgeIface(iface_info2).to_dict()
+        )
+
+    def test_is_master(self):
+        iface = LinuxBridgeIface(gen_bridge_iface_info())
+
+        assert iface.is_master
+
+    def test_is_virtual(self):
+        iface = LinuxBridgeIface(gen_bridge_iface_info())
+
+        assert iface.is_virtual
+
+    def test_get_slaves(self):
+        iface = LinuxBridgeIface(gen_bridge_iface_info())
+
+        assert iface.slaves == TEST_SLAVE_NAMES
+
+    def test_desire_port_name_full_merge_from_current(self):
+        cur_iface_info = gen_bridge_iface_info()
+        des_iface_info = gen_bridge_iface_info()
+        des_iface_info[LB.CONFIG_SUBTREE][LB.PORT_SUBTREE] = [
+            {Port.NAME: SLAVE2_IFACE_NAME},
+            {Port.NAME: SLAVE1_IFACE_NAME},
+        ]
+        des_iface = LinuxBridgeIface(des_iface_info)
+        cur_iface = LinuxBridgeIface(cur_iface_info)
+
+        des_iface.merge(cur_iface)
+
+        assert des_iface.to_dict() == gen_bridge_iface_info()
+
+    def test_desire_port_name_partial_merge_from_current(self):
```

Comment:
I'd prefer the name of the test somehow reflected that we want to preserve one of the two ports. (or remove one of the two ports),

Reply:
Changed to `test_desire_port_name_only_merge_config_from_current`